### DISCUSSION TITLE
ibm_zos_core modules and ZOAU compatibility with JES3 enivironment

### DISCUSSION THREAD

**Discussion by bcronefnts** [ansible-collections/ibm_zos_core#1489] [Q&A] [answered]
I would like to know if the ibm_zos_core modules and IBM ZOAU are compatible with a JES3 environment.  I am getting unpredictable results when attempting to submit a JOB, Query a JOB, or list the OUTPUT of the the JOB.  When submitting a JOB Ansible never gets a return that the JOB completed successfully.  When I attempt to list the OUTPUT of the JOB, the module is successful, but I get no output.  Are these tools compatible with a JES3 environment, or do they only work with JES2?

**Comment by ddimatos** [comment, marked as answer]
Hello @bcronefnts , thank you for your inquiry.

Currently there is no Ansible support for JES3. 
While there is investigation underway by ZOAU for JES3, there is no commitment, thus the soonest Ansible would consider such support would be in Q1 2025.
I will convert this discussion to an issue so that it can be placed in the backlog and also identify system requirements needed for such an item should it be pursued.

**Reply by bcronefnts** [reply]
@ddimatos Thank you for your response. While disappointing, I appreciate the detailed response and moving this to an issue and placed in the backlog for further development.  Thank you again.

**Reply by ddimatos** [reply]
@bcronefnts - thank you, I wish I had better news, this does require a dependency change which technically requires S&S on ZOAU, where support in this community is limited to only the `ibm_zos_core` collection; full stack support has to come through Ansible Automation Platform entitlement, ZOAU and if necessary IBM Open Enterprise SDK for Python.

Beyond that, I can share some more details; we are actually waiting on a change in a dependency so our date is aligned to that planned change, then we have to consider what this means to our offering as it would double some of if not all our current regression to support JES3 (we may be able to limit it to just batch test suites), that regression currently takes 8+ hrs; then there is the issue that none of the systems we currently are using have JES3 (we have 40+ systems), so we would have to plan for that as well. 

While we don't test against JES3, I suspect most our modules would work fine since most don't interact with JES. I appreciate the request, feel free to follow up.

**Comment by ddimatos** [comment]
This discussion has been moved to git issue https://github.com/ansible-collections/ibm_zos_core/issues/1492 and placed in the backlog, continue to follow that issue for updates thus this discussion will be closed.
